src/KeycloakSubscriptionHandler.ts

```diff
@@ -72,20 +71,23 @@ export class KeycloakSubscriptionHandler {
       }
       return
     }
-    const token = this.getBearerTokenFromHeader(header, this.keycloak.config.clientId)
     try {
-      await this.keycloak.grantManager.validateToken(token, 'Bearer')
-      //@ts-ignore
-      return token
+      const accessToken = this.getAccessTokenFromHeader(header)
+      const grant = await this.keycloak.grantManager.createGrant(accessToken)
+      return grant.access_token as unknown as Keycloak.Token
     } catch (e) {
       throw new Error(`Access Denied - ${e}`)
     }
   }
 
-  private getBearerTokenFromHeader(header: any, clientId?: string) {
+  private getAccessTokenFromHeader(header: any) {
     if (header && typeof header === 'string' && (header.indexOf('bearer ') === 0 || header.indexOf('Bearer ') === 0)) {
-      const token = header.substring(7)
-      return new Token(token, clientId)
+      const tokenString = header.substring(7)
+      return {
+        access_token: tokenString
```

Comment:
[Trivial] FieldName is not conforming to standards.

Reply:
Yeah so this is the format required by the `keycloak.grantmanager.createGrant` perhaps it would be better to return just the tokenString value here and then explicitly pass {access_token: tokenString} into that method call to make things clearer.